### DISCUSSION TITLE
Have I've been Pwned password policy.

### DISCUSSION THREAD

**Discussion by honzapatCZ** [zitadel/zitadel#5278] [Ideas]
Policy would not allow using password, which was found in known online breaches.
This could also notify user on login, if their password was leaked in the meantime

**Comment by mffap** [comment]
> Policy would not allow using password, which was found in known online breaches. This could also notify user on login, if their password was leaked in the meantime

Thanks for the suggestion. Prompting on change could be achieved with an Action Flow that triggers on user updates (not yet available).

The login auth flow is available, but I think we don't expose the hashed password, that you need to search for pwnd passwords.

We can take this into consideration when extending Actions. Thanks again.
